### ISSUE TITLE
Avoid rerender if the results from suggestionsCallback is the same

### ISSUE THREAD

**Issue by bee-warrior** [AbdulRahmanAlHamali/flutter_typeahead#84]
Right now the widget rerenders the list every time suggestionsCallback get called. But if the autosuggest results from the backend are the same it would be nice to not re-render the list. Maybe I am missing something? How we can implement it?

**Comment by sjmcdowall** [comment]
I can not think of any clean way to do this without incurring a (large?) cost of comparing the new list with the old list .. and assuming there is a valid comparator for the items in the list .. 

This maybe a "flutter" build() thing -- meaning -- as the Flutter dev team says, expect your build() method to be called a LOT and don't assume anything...   

Have you found this behavior to be horribly inefficient (i.e. noticeably annoying?) ... 

Do you (or anyone) have any ideas of how to avoid this situation?   The only thing I can come up with is to take a return value of null to mean .. don't redraw.. but .. that isn't at all obvious or logical .. Most would assume a null return would be the same as an empty List ...

**Comment by bee-warrior** [comment]
@sjmcdowall @KaYBlitZ 
Well, we use autosuggest every day with Google or Github. Both of them doest rerender the whole list if you are typing the same text from the suggestions list. And from autosuggest design patern, it shouldn't return more than 5-8 items. Comparing between 5-8 items shouldn't be an expensive operation and it will also make ui/ux experience much better

**Comment by KaYBlitZ** [comment]
Flutter works a bit differently. You need a rebuild every time a change is needed, but I think we can limit expensive comparisons  by doing a size check first of the two lists. In this plugin, we don't limit the number of items. It depends on how many items the suggestionsCallback sends back and is up to the programmer. I'll see if I can implement this.

**Comment by KaYBlitZ** [comment]
Actually, nevermind. Took a look at the code and it gets messy real fast if we try to implement this. We have other attributes, such `hideOnLoading` and `loadingBuilder` that requires a rebuild in order to show the loading animation and another rebuild to display the results when done. In order to not have a rebuild we would have to check whether these attributes are being used or not and if we implement this, going forward the code will be unmaintainable due to these edge case checks.

Basically this is trying to fight Flutter itself, where a rebuild is recommended for any state changes. See https://docs.flutter.io/flutter/widgets/State/setState.html . It is up to us to ensure that rebuilds are not expensive, but Flutter almost guarantees that rebuilds are going to be expected and frequent.

**Comment by KaYBlitZ** [comment]
Maybe a possible workaround is disable the sliding animation? The list will still be rebuilding, but will be unnoticeable....

**Comment by KaYBlitZ** [comment]
Closing because I think it is working as intended by Flutter. Feel free to create a new issue if you disagree.
